app/services/card_searcher.rb

```diff
@@ -30,13 +31,22 @@ def only_released_cards
 
   def search_legalities
     @cards = cards.standard_legal if check_standard?
+    @cards = cards.modern_legal if check_modern?
   end
 
   def search_ids
     @cards = cards.where(uuid: params[:ids]) if params[:ids].present?
   end
 
+  def search_sets
+    @cards = cards.where('mtg_sets.code = ?', params[:set]) if params[:set].present?
+  end
+
   def check_standard?
     params[:standard_legal] == 'true'
   end
+
+  def check_modern?
+    params[:modern_legal] == 'true'
+  end
```

Comment:
Go ahead and add filtering for Modern cards in here as well. Will need to come up with a better solution as I add additional legality checks.